toolbox/repo/validate_role_vars_used.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/env python
+
+import glob, os
+import yaml
+import subprocess
+import pathlib
+
+THIS_DIR = pathlib.Path(__file__).absolute().parent
+
+# go to top directory
+os.chdir(os.path.join(THIS_DIR, "..", ".."))
+
+ROLES_VARS_GLOB = "roles/*/vars/*/*"
+ROLES_DEFAULTS_GLOB = "roles/*/defaults/*/*"
+
+def validate_role_vars_used(filename, yaml_doc):
+    errors = 0
+
+    role_name = filename.split("/", 2)[1]
+    for key in yaml_doc.keys():
+        proc = subprocess.run(["grep", key, f"roles/{role_name}",
```

Comment:
You can use `pathlib.Path("roles") / role_name` (division operator) to join path components in a more robust/safe way that string formatting

Reply:
I like it, it reads weird the first time, but when you notice you're not trying to "divide strings", it makes sense!